sql/parser/eval.go

```diff
@@ -535,6 +728,15 @@ var cmpOps = map[cmpArgs]func(Datum, Datum) (Datum, error){
 	cmpArgs{EQ, floatType, floatType}: func(left Datum, right Datum) (Datum, error) {
 		return DBool(left.(DFloat) == right.(DFloat)), nil
 	},
+	cmpArgs{EQ, dateType, dateType}: func(left Datum, right Datum) (Datum, error) {
+		return DBool(left.(DDate).Equal(right.(DDate).Time)), nil
+	},
+	cmpArgs{EQ, timestampType, timestampType}: func(left Datum, right Datum) (Datum, error) {
```

Comment:
Two timestamps representing the same instant with different timezones have different encoded forms, but will compare equal according to this method. This means that a unique index on a timestamp column may have more than one value that compare "equal" to each other. Does this matter to us?